cryptoki/src/context/mod.rs

```diff
@@ -99,23 +100,23 @@ impl Pkcs11 {
                     _pkcs11_lib: pkcs11_lib,
                     function_list: *list_ptr,
                 }),
-                initialized: false,
+                initialized: Arc::new(RwLock::new(false)),
             })
         }
     }
 
     /// Initialize the PKCS11 library
-    pub fn initialize(&mut self, init_args: CInitializeArgs) -> Result<()> {
-        if !self.initialized {
-            initialize(self, init_args)
-        } else {
-            Err(Error::AlreadyInitialized)
+    pub fn initialize(&self, init_args: CInitializeArgs) -> Result<()> {
+        let mut init_lock = self.initialized.as_ref().write().unwrap();
```

Comment:
This is super minor but I'd nonetheless prefer expect here with the reason `lock not to be poisoned` or something like that. I know it's rare but if initialize below panics it'd poison the lock and I'd rather see descriptive error than a generic one in the stacktrace.

Reply:
done.